cmd/guacone/cmd/analyze.go

```diff
@@ -0,0 +1,121 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+  "context"
+  "fmt"
+  "net/http"
+
+  "github.com/Khan/genqlient/graphql"
+  "github.com/guacsec/guac/pkg/logging"
+  "github.com/spf13/cobra"
+  "github.com/spf13/viper"
+  analysis "github.com/guacsec/guac/pkg/analyzer"
+)
+
+const PRINT_MAX = 20
+
+var analyzeCmd = &cobra.Command{
+  Use:     "analyze <operation> <sboms> [flags] ",
+  Short:   "analyze is a CLI tool tailored for comparing, intersecting, and merging Software Bill of Materials (SBOMs) within GUAC",
+  Long:  `Diff Analysis: Compare two SBOMs to identify differences in their software components, versions, and dependencies.
+  Intersection Analysis: Determine the intersection of two SBOMs, highlighting common software components shared between them.
+  Union Analysis: Combine two SBOMs to create a unified SBOM, merging software component lists while maintaining version integrity.`,
+  Example: `
+  Ingest the SBOMs to analyze:
+  $ guacone collect files guac-data-main/docs/spdx/syft-spdx-k8s.gcr.io-kube-apiserver.v1.24.4.json
+  $ guacone collect files guac-data-main/docs/spdx/spdx_vuln.json 
+  Difference
+  $ guacone analyze --diff --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  Union
+  $ guacone analyze --union --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  Intersection
+
+  $ guacone analyze --intersect --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  `,
+  Run: func(cmd *cobra.Command, args []string) {
+    ctx := logging.WithLogger(context.Background())
+    httpClient := http.Client{}
+    gqlclient := graphql.NewClient(viper.GetString("gql-addr"), &httpClient)
+
+
+    //get necessary flags
+    dot, _ := cmd.Flags().GetBool("dot")
+    intersect, _ := cmd.Flags().GetBool("intersect")
+    diff, _ := cmd.Flags().GetBool("diff")
+    union, _ := cmd.Flags().GetBool("union")
+    all, _ := cmd.Flags().GetBool("all")
+    maxprint, _ := cmd.Flags().GetInt("maxprint")
+
+
+    if diff && intersect && union || diff && intersect || diff && union || intersect && union {
+      fmt.Println("Must specify only one of --diff, --intersect, --union")
+      return
+    }
+
+    //create graphs
+    graphs := analysis.HasSBOMToGraph(cmd, ctx, gqlclient)
```

Comment:
Don't pass cmd to a package under `pkg/`. Those shouldn't have a dependency on cli commands or cobra. Explicitly pass the parameters this needs.